### PR TITLE
[ Search ] Link bug fix

### PR DESCRIPTION
**Pull request by Muzm**

### CODE REVIEW

File: packages/helldoc-core/src/app/components/Search.tsx [hunk 1]
```diff
@@ -79,9 +79,12 @@ class Search extends React.Component<{}, SearchState> {
     this.setState({ onFocus: true });
     this.isFullWidth();
   };
-  onLeaveSearchBox = () => {
-    if (isMobileSize()) {
-      this.setState({ fullWidth: false });
+  onLeaveSearchBox = (e: React.FocusEvent) => {
+    if (!e.relatedTarget) {
```
Comment (Nbsaw) [inline packages/helldoc-core/src/app/components/Search.tsx]: Can remove the nested `if` statement

``` js
if (!e.relatedTarget && isMobileSize())
  this.setState({ fullWidth: false });
this.setState({ onFocus: false });
```

or

``` js
const notFullWidth = !e.relatedTarget && isMobileSize()
notFullWidth && this.setState({ fullWidth: false })
this.setState({ onFocus: false });
```

File: packages/helldoc-core/src/app/components/Search.tsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ class Search extends React.Component<{}, SearchState> {
         <input
           ref={this.inputRef}
           type="text"
-          onBlur={this.onLeaveSearchBox}
+          onBlur={e => this.onLeaveSearchBox(e)}
```
Comment (Nbsaw) [inline packages/helldoc-core/src/app/components/Search.tsx]: don't need to pass `e` to function is ok .

see demo https://codesandbox.io/s/l434703q39

**Comment by Nbsaw** [comment]
@fimars LGTM

**Comment by fimars** [comment]
@Muzm @Nbsaw thanks

**Pull request merged**